lib/mongoid/history/attributes/update.rb

```diff
@@ -27,6 +28,22 @@ def insert_embeds_one_changes(relation, value)
           @attributes[relation][1] = value[1][paranoia_field].present? ? {} : format_embeds_one_relation(relation, value[1])
         end
 
+        def insert_embeds_one_changes_on_child
```

Comment:
Should be renamed and refactored to return changes on embeds. The code for non-embeds should be changed to return changes on non-embeds, finally `attributes` should collect all of the things.